### PR TITLE
CMC-1885: Inform agreed extension date - 1v2 diff solicitors

### PR DESCRIPTION
**Pull request by jeejavr**
https://tools.hmcts.net/jira/browse/CMC-1885

Added code to validate extension date entered by second defendant solicitor in 1v2 scenario
Added code to restrict users from accessing the event again
Code updation to reflect the extended date of respondent 2 on the confirmation page
Test cases updation
Added tests to cover new code changes

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

**Review by akriti-hmcts — COMMENTED** [review]
There are failing tests on this PR. @jeejavr Please fix those

**Review by akriti-hmcts — COMMENTED** [review]
There is no unit test added for the scenario on InformAgreedExtensionDateCallbackHandler

File: src/test/java/uk/gov/hmcts/reform/civil/handler/callback/user/InformAgreedExtensionDateCallbackHandlerTest.java [hunk 1]
```diff
@@ -159,6 +161,98 @@ void shouldSetRespondent1FlagToNo_whenTwoRespondentRepresentativesWithRespondent
...
+                .respondent1TimeExtensionDate(timeExtensionDate)
+                .build();
+
+            CallbackParams params = callbackParamsOf(V_1, caseData, ABOUT_TO_START);
+
+            AboutToStartOrSubmitCallbackResponse response = (AboutToStartOrSubmitCallbackResponse) handler
+                .handle(params);
+
+            assertThat(response.getErrors())
+                .containsOnly("This action cannot currently be performed because it has already been completed");
```
Comment (akriti-hmcts) [inline src/test/java/uk/gov/hmcts/reform/civil/handler/callback/user/InformAgreedExtensionDateCallbackHandlerTest.java:234]: Can this String be defined as a final variable please

**Review by akriti-hmcts — CHANGES_REQUESTED** [review]

**Review by akriti-hmcts — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/civil/service/robotics/mapper/EventHistoryMapper.java [hunk 2]
```diff
@@ -671,18 +675,37 @@ private void buildConsentExtensionFilingDefence(EventHistory.EventHistoryBuilder
...
+                && (caseData.getRespondent2TimeExtensionDate() != null)) {
+                if (caseData.getRespondent2TimeExtensionDate()
+                    .isAfter(caseData.getRespondent1TimeExtensionDate())) {
+                    extensionDate = caseData.getRespondentSolicitor2AgreedDeadlineExtension();
+                } else {
+                    extensionDate = caseData.getRespondentSolicitor1AgreedDeadlineExtension();
+                }
+            }
+        }
+
```
Comment (HarryH96) [inline src/main/java/uk/gov/hmcts/reform/civil/service/robotics/mapper/EventHistoryMapper.java:697]: Test cases will need to be added for this file.
In relation to the `dateReceived` field in this method, it is currently specific to respondent 1 only.. What if respondent2 is asking for a time extension, is this date being stored somewhere?

**Review by akriti-hmcts — APPROVED** [review]

**Review by mounikahmcts — APPROVED** [review]

**Review by douglasrice — APPROVED** [review]

**Pull request merged**